### ISSUE TITLE
Show detailed info about future payments

### ISSUE THREAD

**Issue by ntnsndr** [gratipay/inside.gratipay.com#993]
@whit537 edit:

In https://github.com/gratipay/gratipay.com/pull/4128 we're working on a history page for projects that will give detailed info about past transactions (though they'll all be anonymous until https://github.com/gratipay/gratipay.com/issues/236). Because of the way we charge in arrears (batching up small payments until $10 is reached), actual receiving can be wildly different from expected receiving. We give a minimal expectation of future receiving with the Funded and Unfunded Dues numbers on the project receiving page, but those are aggregate numbers, and it takes some education to understand that they exist and what they mean. The proposal here is to add a detailed view of future expected transactions, to include amount and date.

-----------------

### Original

Friends, we've got a problem with the Internet of Ownership Gratipay account. The payments we're getting are coming to about $8.05 per month:

![](https://loomio-attachments.s3.amazonaws.com/attachments/files/000/099/788/original/Screenshot_2017-01-20_at_13.43.30.png?1484948629&utm_campaign=new_comment&utm_medium=email&utm_source=thread_mailer)

While, according to Gratipay itself, we should be receiving $11.08 per *month*. Can you explain the discrepancy?

Thanks for all your good work.

**Comment by nobodxbodon** [comment]
Thanks a lot @ntnsndr for the report. To say the least, this is very confusing, even if there's no issue/bug related in the code or procedure. May I ask if this is the only occurrence you noticed? If I recall correctly there's a history page under your account that you may check all the transactions.

**Comment by mattbk** [comment]
Looking into it.

**Comment by whit537** [comment]
Initial hunch is that it has to do with charging in arrears. If someone gives 25¢/wk then we batch that up until it reaches $10 before charging. Might that explain it?

**Comment by whit537** [comment]
I see "funded dues" approaching $50 on https://gratipay.com/internet-of-ownership/receiving/.

**Comment by mattbk** [comment]
Minus ~1.9% PayPal fees per payout.

**Comment by nobodxbodon** [comment]
If it's caused by fees and batch-up, IMO we need to make it as clear as possible to users, like https://github.com/gratipay/gratipay.com/issues/3326.

Let's get to the bottom of this case first. Pitifully I don't have access to https://gratipay.com/internet-of-ownership/receiving/. Maybe I shall check logs?

**Comment by mattbk** [comment]
And as @whit537 says, arrears.

> Estimated payment for next week: $15.25

**Comment by ntnsndr** [comment]
That doesn't really seem adequate to explaining why we are getting less than a quarter of what we appear to be owed.

**Comment by mattbk** [comment]
Not just the fees. There's almost $50 that will pay out once people's credit cards are charged. Next week $15 is an example of this. 

It's very hard to parse out a specific case without someone looking at the database directly; very few of us have that access so this may have to wait a few days. 

@nobodxbodon, I'm surprised you don't have admin access.

**Comment by nobodxbodon** [comment]
> That doesn't really seem adequate to explaining why we are getting less than a quarter of what we appear to be owed.

@ntnsndr do you refer to the weekly receiving shown on project page?
![screen shot 2017-01-20 at 8 58 42 pm](https://cloud.githubusercontent.com/assets/392497/22172086/48a0c644-df53-11e6-93f5-71b5317de6f1.png)

> we should be receiving $11.08 per month

So I guess you meant "per _week_" here instead of "per _month_"?

@mattbk @whit537 by looking at the report it seems payments is received by month, instead of by week. I thought we run payday per week?

Besides, IMO we should provide a "receiving expected" report other than the [history page](https://github.com/gratipay/gratipay.com/pull/4128), so that user can better forecast future receiving.

**Comment by mattbk** [comment]
Payments are intermittent (monthly) for them because of arrears. This is one of my main gripes with the system as it is now.

**Comment by whit537** [comment]
Okay ... I checked in the database, and what I see does match up with the payouts that are in your screenshot, @ntnsndr. So we think we sent what you actually received. As to why we only sent as much as we did ...

**Comment by whit537** [comment]
I find 54 payments from three people since last March.

```sql
  select "timestamp", participant, amount
    from payments
   where team ='internet-of-ownership' and direction='to-team'
order by timestamp desc;
```

**Comment by whit537** [comment]
Now, I find 16 payment _instructions_:

```sql
  select amount, due, is_funded
    from current_payment_instructions
   where team_id=247
order by is_funded desc, due desc;
```

```
┌────────┬──────┬───────────┐
│ amount │ due  │ is_funded │
├────────┼──────┼───────────┤
│   1.00 │ 6.00 │ t         │
│   2.00 │ 6.00 │ t         │
│   1.00 │ 6.00 │ t         │
│   1.00 │ 6.00 │ t         │
│   1.00 │ 6.00 │ t         │
│   1.00 │ 6.00 │ t         │
│   1.00 │ 6.00 │ t         │
│   0.30 │ 1.80 │ t         │
│   0.25 │ 1.50 │ t         │
│   0.25 │ 1.50 │ t         │
│   0.22 │ 1.32 │ t         │
│   0.01 │ 0.03 │ t         │
│   0.25 │ 0.00 │ t         │
│   2.00 │ 0.00 │ t         │
│   1.00 │ 0.00 │ f         │
│   0.25 │ 0.00 │ f         │
└────────┴──────┴───────────┘
(16 rows)
```

**Comment by whit537** [comment]
The "$11.08 per [_week_]" [number](https://github.com/gratipay/inside.gratipay.com/issues/993#issue-202251875) that is displayed on IoO's [profile page](https://gratipay.com/internet-of-ownership/) ...

![screen shot 2017-01-23 at 3 12 04 pm](https://cloud.githubusercontent.com/assets/134455/22220852/68dc093c-e17e-11e6-95ce-af07eb879b22.png)

... is coming from this `current_payment_instructions` view, and is the sum of the `is_funded` `amount`s:

```sql
  select sum(amount) as amount, sum(due) as due, count(amount) as n, is_funded
    from current_payment_instructions
   where team_id=247
group by is_funded
order by is_funded desc;
```

```
┌───────┬───────┬───────┬───────────┐
│  sum  │  sum  │ count │ is_funded │
├───────┼───────┼───────┼───────────┤
│ 11.28 │ 48.15 │    14 │ t         │
│  1.25 │  0.00 │     2 │ f         │
└───────┴───────┴───────┴───────────┘
(2 rows)
```

**Comment by whit537** [comment]
@ntnsndr So the good news is the Gratipay is behaving as intended—there does not seem to be any database corruption or snafu between us and PayPal or egregious bug in our billing code. However, our behavior is certainly confusing, for which I apologize.

When a giver is giving less than $10 (in aggregate to all projects on Gratipay), we keep track of their weekly pledge until they reach $10, and _then_ we charge them (this is what "charging in arrears" means [above](https://github.com/gratipay/inside.gratipay.com/issues/993#issuecomment-274216167), and it's done to minimize credit card processing fees). So basically you should expect a series of larger payouts once those [smaller $1 and $2 payments](https://github.com/gratipay/inside.gratipay.com/issues/993#issuecomment-274601869) accrue to $10, which should happen in ... four weeks or so, it looks like.

Am I making it any clearer what is going on, @ntnsndr?

If we're on the same page as far as the existing reality goes, then let's talk about ways to possibly alter Gratipay's behavior or at least communicate it more clearly.

One way or another, sorry for the confusion. 😞

**Comment by ntnsndr** [comment]
Oh, yes, I see now. @devinbalkind, see this? 

Maybe this should be made more transparent on the platform when payments are made? So we know how much is being held and is owed?

**Comment by mattbk** [comment]
> is owed

That's what funded dues is at https://gratipay.com/internet-of-ownership/receiving/. Money you are due to receive that has a working credit card behind it.

**Comment by devinbalkind** [comment]
Yes I'd recommend having a transactions table that shows the date and amount of previous transactions and another table that shows future transactions (ceteris paribus) so people get an idea of what to expect.

**Comment by whit537** [comment]
The date and amount of previous transactions will be available once we land [project history pages](https://github.com/gratipay/gratipay.com/pull/4128) (though they'll all be anonymous until we start [breaking anonymity](https://github.com/gratipay/gratipay.com/issues/236)—a high priority). Unpacking the "funded dues" number is another task. We may already have a ticket for that, but until we locate it let's use this for that.

Longer term, if/when we can get around to [offering recurring schedules other than weekly](https://github.com/gratipay/gratipay.com/issues/3862), we'll be in a position to drop charging in arrears entirely. At that point we can simply make $1 the minimum payment, and always charge no matter what (the giver pays the fee, so they'd be charged $1.36 or whatever &hellip; once a month, once a year—up to them).

@ntnsndr @devinbalkind How would you like to use this issue?

1..

**Comment by whit537** [comment]
(Blech, sorry for the comment fart. Edited.)

**Comment by whit537** [comment]
Another possibility would be to drop the minimum charge from $10 to $1 right now. That would dampen the variability that comes from charging in arrears, perhaps to an acceptable level. Thoughts?

**Comment by mattbk** [comment]
(From a support perspective, yes please. Should be the same as long as the masspay transaction fee is a pct, yes? )

**Comment by devinbalkind** [comment]
I like the idea of reducing transaction costs with the current approach and don't mind receiving payments over a longer frequency (monthly works for me). I just want to know what to expect so I know if things are going wrong.

**Comment by whit537** [comment]
> but what about fees?

Fees would be proportionally higher, $1.36 = 26% vs. $10.61 = 6% (or whatever).

> I like the idea of reducing transaction costs with the current approach and don't mind receiving payments over a longer frequency (monthly works for me). I just want to know what to expect so I know if things are going wrong.

Okay. Let's keep things as they stand, then, and look for better ways to communicate. We do have a [note in the FAQ](https://gratipay.com/about/features/#receiving-less), fwiw.

**Comment by mattbk** [comment]
> Besides, IMO we should provide a "receiving expected" report other than the history page, so that user can better forecast future receiving.

@nobodxbodon, this is what "Estimated payment for next week:" is on the receiving page.

**Comment by nobodxbodon** [comment]
> @nobodxbodon, this is what "Estimated payment for next week:" is on the receiving page.

IMO it's not enough to show next week's estimate (in this case would be 0 in 3 out of 4 weeks in a month for the past year). At least we should show the next payment if there's any, and ideally show all the future payments based on current payment instructions.

**Comment by devinbalkind** [comment]
Yes I think 5 weeks of future payments would be a good amount.
